### DISCUSSION TITLE
Delete Body vs /delete POST endpoint vs....

### DISCUSSION THREAD

**Discussion by adamkorynta** [USACE/cwms-data-api#754] [General Aesthetics]
The REST specification does not include a body for the Delete HTTP method. If we want to perform bulk delete operations, I see suggestions to either put a content body in anyways, or create a new endpoint and perform the action via POST and rely on naming. There's plenty of back-and-forth discourse on there on the topic. Wondering if CDA developers have a preference.

## Ex. 1 - Delete Body
`DELETE /turbine/changes/<location>?office=<office>`
```
{ [
     2020-01-01T00:00:00Z,
     2020-01-01T01:00:00Z,
     2020-01-01T02:00:00Z,
     2020-01-01T03:00:00Z,
     2020-01-01T04:00:00Z,
]} 
```

## Ex. 2 - Post Delete Endpoint
`POST /turbine/changes/bulk-delete/<location>?office=<office>`
```
{ [
     2020-01-01T00:00:00Z,
     2020-01-01T01:00:00Z,
     2020-01-01T02:00:00Z,
     2020-01-01T03:00:00Z,
     2020-01-01T04:00:00Z,
]} 
```

## Time Window Delete
We'd still want to include a time window delete (required by java client), but I'm guessing a bulk delete would be useful to a broader audience.

`DELETE /turbine/changes/<location>?office=<office>&start=<start-date>&end=<end-date>`

**Comment by rma-rripken** [comment]
My preference is that we implement the expected/least surprising thing and then add any extra cool features on top.  So a boring single item delete that uses http.DELETE and then if you want a bulk delete or some other operation it sits at its own url and uses POST.

**Reply by MikeNeilson** [reply]
We could also support both. the no body in delete is really an HTTP thing but I don't think i've found an HTTP server that doesn't support it.
